### PR TITLE
ESP32: add board esp32nick

### PR DESCRIPTION
**Pull request by nkruzan**
This adds support for a esp32 based board esp32nick. Nothing groundbreaking or revolutionary, just a board I made at home. and test with.

### CODE REVIEW

File: libraries/AP_HAL/board/esp32.h [hunk 1]
```diff
@@ -11,6 +11,8 @@
 #include "esp32empty.h" //wiktor-m
 #elif CONFIG_HAL_BOARD_SUBTYPE == HAL_BOARD_SUBTYPE_ESP32_TOMTE76
 #include "esp32tomte76.h" //tomte76 on discord
+#elif CONFIG_HAL_BOARD_SUBTYPE == HAL_BOARD_SUBTYPE_ESP32_NICK
+#include "esp32nick.h" //tomte76 on discord
```
Comment (davidbuzz) [inline libraries/AP_HAL/board/esp32.h:15]: incorrect comment

**Review by davidbuzz — APPROVED** [review]
looks good to me.

**Comment by davidbuzz** [comment]
good to merge  after u fix that one comment.

File: libraries/AP_HAL/board/esp32.h (hunk 1 above)
Comment (nkruzan) [reply libraries/AP_HAL/board/esp32.h]: fixed, thanks

**Pull request merged**